### ISSUE TITLE
cifar10 data visualization

### ISSUE THREAD

**Issue by jprachir** [clementchadebec/benchmark_VAE#52]
Hi @clementchadebec,
The following code loads the CIFAR10 data as NumPy train & eval:

```
cifar10_trainset = datasets.CIFAR10(root='../../data', train=True, download=True, transform=None)
# array
train_dataset = cifar10_trainset.data[:-10000].reshape(-1, 3, 32, 32) #(40k,3,32,32)
eval_dataset = cifar10_trainset.data[-10000:].reshape(-1, 3, 32, 32) # (10k,3,32,32)
```
when I try to visualize, why it's not an image from the CIFAR10 dataset, what am I doing wrong here?

```
npimg = train_dataset[0] # first image from training data
img_ar = np.transpose(npimg, (1,2,0))
plt.imshow(img_ar)
```
![image](https://user-images.githubusercontent.com/20530658/188295498-d6d39f5e-b03f-46f0-a574-2bf4a2a95adf.png)

*also I can't assign the label (it gets hidden) when I create new issues.

Thanks,
Prachi

**Comment by clementchadebec** [comment]
Hi @jprachir,
I think the `reshape` are the cause of the issue. You can use `np.moveaxis` instead.
 
Can you try loading the database with this:
```python
cifar10_trainset = datasets.CIFAR10(root='../../data', train=True, download=True, transform=None)
# array
train_dataset = np.moveaxis(cifar10_trainset.data[:-10000], 3, 1) #(40k,3,32,32)
eval_dataset = np.moveaxis(cifar10_trainset.data[-10000:], 3, 1) # (10k,3,32,32)
```
and then visualize an image running this:
```python
npimg = train_dataset[1] # first image from training data
img_ar = np.moveaxis(npimg, 0, 2)
plt.imshow(img_ar)
```

I think this should do the trick :)

Let me know if you have any other questions.

Best,

Clément

**Comment by jprachir** [comment]
Hi @clementchadebec,
This certainly worked out. 

shapes & format:
cifar10_trainset returns the tuples of **(PIL_image, int_target)** & it's size is (50k, 3, 32, 32)  in the format (N,C,H,W) 
```
npimg = train_dataset[1] # (3,32,32) := (C,H,W) first image from training data 
img_ar = np.moveaxis(npimg, 0, 2) # (32,32,3) := (H,W,C) moving axis 
plt.imshow(img_ar)
```
![image](https://user-images.githubusercontent.com/20530658/188509532-1f8745fc-a622-4d50-825f-92347bb80188.png) 
It's a truck!

Thanks,
Prachi

**Comment by clementchadebec** [comment]
No problem! Glad it worked :) 
I am closing this since it appears to be resolved. Do not hesitate to reopen it if needed.
Cheers
